File: ringstrings-tests/src/main/java/com/coreman2200/ringstrings/symbol/astralsymbol/grouped/CelestialBodies.java
package com.coreman2200.ringstrings.symbol.astralsymbol.grouped;

/**
 * CelestialBodies
 * Symbols for considered Celestial Bodies
 *
 * Created by Cory Higginbottom on 5/29/15
 * http://github.com/coreman2200
 *
 * Licensed under the GNU General Public License (GPL), Version 2.0.
 * You may not use this file except in compliance with this License.
 *
 * You may obtain a copy of the GPLv2 License at
 * http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 */

public enum CelestialBodies {
    SUN(0),
    MOON(1),
    MERCURY(2),
    VENUS(3),
    MARS(4),
    JUPITER(5),
    SATURN(6),
    URANUS(7),
    NEPTUNE(8),
    PLUTO(9),
    NORTHNODE(11),
    LILITH(13),
    EARTH(14),
    CHIRON(15),
    CERES(17),
    PALLAS(18),
    JUNO(19),
    VESTA(20),
    ASCENDANT(98),
    MIDHEAVEN(99);

    int mSwissephValue;

    CelestialBodies(int val) {
        this.mSwissephValue = val;
    }

    public int getSwissephValue() {
        return this.mSwissephValue;
    }
}

File: domain/src/main/java/com/coreman2200/ringstrings/domain/symbol/astralsymbol/grouped/CelestialBodies.kt
package com.coreman2200.ringstrings.domain.symbol.astralsymbol.grouped

import com.coreman2200.ringstrings.domain.symbol.symbolinterface.ISymbolID

/**
 * CelestialBodies
 * Symbols for considered Celestial Bodies
 *
 * Created by Cory Higginbottom on 5/29/15
 * http://github.com/coreman2200
 *
 * Licensed under the GNU General Public License (GPL), Version 2.0.
 * You may not use this file except in compliance with this License.
 *
 * You may obtain a copy of the GPLv2 License at
 * http://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 */
enum class CelestialBodies(val swissephValue: Int, val isRealCelestialBody: Boolean) : ISymbolID {
    SUN(0, true), MOON(1, true), MERCURY(2, true), VENUS(3, true), MARS(4, true), JUPITER(
        5,
        true
    ),
    SATURN(6, true), URANUS(7, true), NEPTUNE(8, true), PLUTO(9, true),
    NORTHNODE(
        11,
        true
    ),
    SOUTHNODE(100, false), LILITH(
        13,
        true
    ), // EARTH(14),
    CHIRON(15, true), CERES(17, true), PALLAS(18, true), JUNO(19, true), VESTA(20, true), ASCENDANT(
        98,
        false
    ),
    MIDHEAVEN(99, false);
}
